pkg/kubelet/kubelet.go

```diff
@@ -1715,6 +1727,38 @@ func (kl *Kubelet) syncLoop(updates <-chan PodUpdate, handler SyncHandler) {
 	}
 }
 
+// syncStatus syncs pods statuses with the apiserver.
+func (kl *Kubelet) syncStatus() {
+	glog.V(3).Infof("Syncing pods status")
+
+	statuses := make(map[string]api.PodStatus)
+	kl.podLock.Lock()
```

Comment:
We usually always use defer to reduce the probability of bugs. For cases like this we use this pattern:

```Go

statuses := ...
func() {
  lock.Lock()
  defer lock.Unlock()
}

for podFullName, ... {
}
```